packages/react-hooks/src/useAsync.ts

```diff
@@ -1,15 +1,19 @@
 import { Async } from '@fluentui/utilities';
 import * as React from 'react';
-import { useConst } from './useConst';
 
 /**
  * Hook to provide an Async instance that is automatically cleaned up on dismount.
  */
 export function useAsync() {
-  const async = useConst<Async>(() => new Async());
-
-  // Function that returns a function in order to dispose the async instance on unmount
-  React.useEffect(() => () => async.dispose(), [async]);
-
-  return async;
+  const asyncRef = React.useRef<Async>();
+  if (!asyncRef.current) {
+    asyncRef.current = new Async();
+  }
+  React.useEffect(() => {
+    return () => {
+      asyncRef.current?.dispose();
+      asyncRef.current = undefined;
+    };
+  }, []);
```

Comment:
This effect has no dependencies, so I'm not sure if we will hit a state where dispose is called, but then useAsync creates a new instance, and then that never gets disposed.

Reply:
the 2nd useAsync will be disposed just like the first one, as soon as the component unmounts and the return function is called. Adding a ref to a callback is unnecessary because the callback always refers to the current value of the ref.